src/module/actor/creature/sheet.ts

```diff
@@ -68,13 +67,13 @@ export abstract class CreatureSheetPF2e<TActor extends CreaturePF2e> extends Act
 
         // Update skill labels
         if (sheetData.data.skills) {
-            const skills: Record<string, SkillData & Record<string, string>> = sheetData.data.skills;
-            const mainSkills: Record<string, string> = CONFIG.PF2E.skills;
-            for (const key in skills) {
-                const skill = skills[key];
+            const skills = sheetData.data.skills;
+            for (const [key, skill] of Object.entries(skills)) {
+                if (!skill.rank) continue;
```

Comment:
Some skills have no rank?

Reply:
NPC skill are supposed to have no rank but the prepared skill data still has it with a value of 1. I wasn't sure if every `Creature` subclass defines a skill rank so I thought it would be better to add a check. 
Now that I look at the old code again it shouldn't ever be undefined or it would have blown up already.